Tests/RedisKitTests/RedisKitTests.swift

```diff
@@ -31,6 +151,263 @@ final class RedisKitTests: XCTestCase {
+        XCTAssertEqual(rpushResp, 2)
+
+        let length = try self.client.llen(of: "mylist").wait()
+        XCTAssertEqual(length, 2)
+
+        let item = try self.client.lindex(0, from: "mylist").wait()
+        XCTAssertEqual(item.string, "hello")
+        
+        let items = try self.client.lrange(within: (startIndex: 0, endIndex: 1), from: "mylist").wait()
+        XCTAssertEqual(items.count, 2)
+
+        try self.client.lset(index: 0, to: RESPValue(bulk: "hello2"), in: "mylist").wait()
+        let item2 = try self.client.lindex(0, from: "mylist").wait()
+        XCTAssertEqual(item2.string, "hello2")
+
+        let rpopResp = try self.client.rpop(from: "mylist").wait()
+        XCTAssertEqual(rpopResp.string, "hello1")
+
+        let rpoplpush = try self.client.rpoplpush(from: "mylist", to: "list2").wait()
+        XCTAssertEqual(rpoplpush.string, "hello2")
+
+        let lpopResp = try self.client.lpop(from: "list2").wait()
+        XCTAssertEqual(lpopResp.string, "hello2")
+
+        let blpopResp1 = try self.client.blpop(from: ["mylist"], timeout: 1).wait()
+        XCTAssertNil(blpopResp1)
+
+        let _ = try self.client.lpush([RESPValue(bulk: "hello")], into: "mylist").wait()
+        let blpopResp2 = try self.client.blpop(from: ["mylist"], timeout: 1).wait()
+        XCTAssertEqual(blpopResp2?.0, "mylist")
+        XCTAssertEqual(blpopResp2?.1.string, "hello")
+
+        let brpopResp1 = try self.client.brpop(from: ["mylist"], timeout: 1).wait()
+        XCTAssertNil(brpopResp1)
+
+        let _ = try self.client.lpush([RESPValue(bulk: "hello")], into: "mylist").wait()
+        let brpopResp2 = try self.client.brpop(from: ["mylist"], timeout: 1).wait()
+        XCTAssertEqual(brpopResp2?.0, "mylist")
+        XCTAssertEqual(brpopResp2?.1.string, "hello")
+
+        let brpoplpushResp1 = try self.client.brpoplpush(from: "mylist", to: "list2", timeout: 1).wait()
+        XCTAssertNil(brpoplpushResp1)
+
+        let _ = try self.client.lpush([RESPValue(bulk: "hello")], into: "mylist").wait()
+        let brpoplpushResp2 = try self.client.brpoplpush(from: "mylist", to: "list2", timeout: 1).wait()
+        XCTAssertEqual(brpoplpushResp2!.string, "hello")
+        let brpoplpushResp3 = try self.client.lpop(from: "list2").wait()
+        XCTAssertEqual(brpoplpushResp3.string, "hello")
+
+        let _ = try self.client.lpush([RESPValue(bulk: "hello"), RESPValue(bulk: "hello1"), RESPValue(bulk: "hello")], into: "mylist").wait()
+
+        XCTAssertEqual(try self.client.llen(of: "mylist").wait(), 3)
+        XCTAssertEqual(try self.client.lrem("hello", from: "mylist", count: 1).wait(), 1)
+        XCTAssertEqual(try self.client.llen(of: "mylist").wait(), 2)
+
+        let _ = try self.client.delete(["mylist", "list2"]).wait()
+    }
+
+    func testExpire() throws {
+        #warning("client.expire() uses TimeAmount which is deprecated in nio")
```

Comment:
Hmm I wonder what NIO wants us to use instead? We should open an issue for this here.

Reply:
just a regular Int if I'm not mistaken, but this should be fixed on the redistack side I think

Reply:
Yes, this should be fixed in RediStack. I wonder when this got deprecated.

Could you file an issue against it?

https://GitLab.com/mordil/Swift-Redi-stack/issues

Reply:
After looking into this a bit more, I don't think this is accurate and a non-issue (I'll be updating the ticket in RediStack).

The only part of `TimeAmount` that is deprecated is the `typealias Value = Int64`.

Given that `expire` expects `TimeAmount`, the callsite on Line 319 should read:

```swift
let _ = try self.client.expire("foo", after: .seconds(1)).wait()
```

which is still perfectly valid as of NIO 2.9